### ISSUE TITLE
Scheduled and deadline-d TODOs still show up in the agenda when done 

### ISSUE THREAD

**Issue by akinsho** [kristijanhusak/orgmode.nvim#30]
**Describe the bug**
When completing a todo that has a deadline or schedule, despite refreshing I still see the task in my agenda for the week, which gets noisy and confusing. I believe that normally in org these tasks would disappear (could be wrong).

**To Reproduce**
1. Go to an incomplete todo with a deadline or scheduled date
2. Click on complete the task
3. Open agenda view
4. See task still in the agenda

**Expected behavior**
If in a completed state, either done or a user defined state, a task should no longer appear in the agenda.

**Emacs functionality**
Based on the docs [here](https://orgmode.org/guide/Deadlines-and-Scheduling.html#Deadlines-and-Scheduling) I believe that when a todo item is completed the agenda should no longer flag it.

**Screenshots**
![image](https://user-images.githubusercontent.com/22454918/124563354-def38180-de37-11eb-9e2e-800b8bcb1e1e.png)
The cancelled and done task in the screenshot above are still visible under yesterday's heading

**Comment by kristijanhusak** [comment]
Actually that's the default behavior. Here's screenshot from emacs:

![screenshot](https://i.imgur.com/TgifmnQ.png)

But it is configurable according to [this SO comment](https://stackoverflow.com/a/8282185/1474465) and [orgmode manual](https://orgmode.org/manual/Deadlines-and-Scheduling.html#index-org_002dagenda_002dskip_002dscheduled_002dif_002ddone-817)

So I'll convert this to a feature issue.

**Comment by akinsho** [comment]
Awesome thanks for clarifying @kristijanhusak, I noticed that some were being removed but that SO post clarifies that it was the ones I was marking as archived that were being removed, but done on it's own by default doesnt do that.

An option for this would be fantastic 👍🏾

**Comment by kristijanhusak** [comment]
I added options for this:

Hide scheduled: https://github.com/kristijanhusak/orgmode.nvim/blob/master/DOCS.md#org_agenda_skip_scheduled_if_done

Hide deadline: https://github.com/kristijanhusak/orgmode.nvim/blob/master/DOCS.md#org_agenda_skip_deadline_if_done

**Comment by jmrussell** [comment]
The hide deadline option works for me but the hide schedule option doesn't seem to make a difference when set to true.

**Comment by kristijanhusak** [comment]
@jmrussell For me it works fine. How exactly are you testing it?

**Comment by jmrussell** [comment]
Hi @kristijanhusak ,

Attached are a screenshot of my agenda and my config for orgmode. 

![image](https://user-images.githubusercontent.com/8886267/145255696-ca0fc6a3-b932-4b82-a5d9-d41fc447a57a.png)

![image](https://user-images.githubusercontent.com/8886267/145255878-f002c88e-246f-4c4f-969b-14666c0630a2.png)


I'm probably missing something simple... but as I understand it I shouldn't see the "DONE" tasks in my agenda mode with that config. Great plugin by the way, I am really enjoying using it.

**Comment by kristijanhusak** [comment]
@jmrussell I figured out whats the issue. You don't have a separator in the `org_todo_keywords` that separates the `todo` keywords from `done` keywords. If there is no separator, last item is considered as a `done` type, so you didn't do anything wrong here. There was a bug with `fast access` and missing separator, but I pushed the fix just now, so if you pull latest master and give it a try, it should properly hide done types even without the `org_agenda_skip_scheduled_if_done` option.
